src/system_simulation/trace_visualization.rb

```diff
@@ -83,15 +89,18 @@ def add_circle_trace(node_id, _sampling_rate, stats)
 
       curve_points << position
 
-      # only plot dots for first period, after that only draw the curve line
-      break if current_data_sample.time_stamp.to_f >= trace_time_limit
+      # only plot dots within the largest amplitude, after that only draw the curve line
+      next unless index >= start_index && index <= end_index
 
       # distance to last point basically representing the speed (since time interval between data samples is fixed)
       distance_to_last = position.distance(last_position)
+      # max_distance can be zero for non moving nodes
       distance_ratio = (distance_to_last / max_distance)
+      distance_ratio = 0 if distance_ratio.nan? || distance_ratio.infinite?
 
       # invert distance ratio since high distance should plot a small and lightly colored dot
       ratio = 1 - distance_ratio
+      Geometry.clamp(ratio, 0.0, 1.0)
```

Comment:
This is a NOOP, as the ratio is not written back.
Should be `ratio = Geometry.clamp(...)`

Reply:
That is true, good catch :-)

Reply:
Fixed in c97e79f